PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc

```diff
@@ -435,10 +539,112 @@ void pat::PATIsolatedTrackProducer::getIsolation(const LorentzVector& p4,
 
 }
 
+//get overlap of isolated track with a PF lepton
+bool pat::PATIsolatedTrackProducer::getPFLeptonOverlap(const LorentzVector& p4, const pat::PackedCandidateCollection *pc, int pc_idx) const
+{
+        bool isOverlap = false;
+	float dr_min = pflepoverlap_DR_;
+	int id_drmin = 0;
+        for(pat::PackedCandidateCollection::const_iterator pf_it = pc->begin(); pf_it != pc->end(); pf_it++){
+//	    // Count candidate itself (commenting out lines below): exclude isolated track that is a pf lepton
+//            if(int(pf_it - pc->begin()) == pc_idx)
+//                continue;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
I guess one can argue that the inline comment is in place

Reply:
it looks like pc_idx argument and this commented out piece can be removed

Reply:
@mmasciov just to be sure, the intended behaviour is that isolated tracks that are pf candidates that are leptons should always fail `getPFLeptonOverlap`, right? (provided they pass fromPV)

Reply:
@slava77 : done

Reply:
@gpetruc : that is correct, this is why I wanted to count the isolated track itself.